### PR TITLE
docs: port docs source to markdown, update all links

### PR DESCRIPTION
**Pull request by CalMacCQ**
# Description

Same as https://github.com/CQCL/pytket-quantinuum/pull/522 but for pytket-qiskit

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

**Review by cqc-melf — COMMENTED** [review]
Looks good. Do you plan to do this for the other extensions as well?

Does this solves the issues as well we had on the release workflow with the docs?

**Comment by CalMacCQ** [comment]
Ah think I might have messed up some changes to `ibm.py` made in the `main` branch.

File: pytket/extensions/qiskit/backends/ibm.py [hunk 1]
```diff
@@ -75,17 +75,14 @@
 from pytket.utils import prepare_circuit
 from pytket.utils.outcomearray import OutcomeArray
 from pytket.utils.results import KwargTypes
-from qiskit.primitives import (  # type: ignore
-    BitArray,
-    DataBin,
-    PrimitiveResult,
```
Comment (CalMacCQ) [inline pytket/extensions/qiskit/backends/ibm.py:81]: Slightly concerned that I messed this file up when merging in `main`. Does it look okay to you @cqc-alec ?

Comment (CalMacCQ) [reply pytket/extensions/qiskit/backends/ibm.py:81]: In fact, I think its fine... Any discrepancy with main should manifest as a merge coinflcit.

Comment (cqc-alec) [reply pytket/extensions/qiskit/backends/ibm.py:81]: No this is not right, looks like you've reverted the changes in #407 .

Comment (CalMacCQ) [reply pytket/extensions/qiskit/backends/ibm.py:81]: Sorry, will fix.

Comment (CalMacCQ) [reply pytket/extensions/qiskit/backends/ibm.py]: [c347b2a](https://github.com/CQCL/pytket-qiskit/pull/405/commits/c347b2a277cb8d637ad6179065fee3de09491899)

**Pull request merged**